pkg/controller/cluster/cluster_reconcile.go

```diff
@@ -63,6 +68,23 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (reconcile.Resul
 	repoName := request.NamespacedName
 	repoName.Name = instance.Spec.TenantRef.Name
 
+	client, err := vault.NewClient()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	token, err := r.getServiceAccountToken(instance)
+	if err != nil {
+		if err.Error() != "no token available yet" {
```

Comment:
Please create a proper error instead of string comparison

Reply:
Also this should still `return`, right?
Thinking about it, I don't think this should be a special case. It's just an error and should be handled like one...

Reply:
As discussed we simply skip this case and return the first found secret with a token.